lib/compat.php

```diff
@@ -8,6 +8,104 @@
  * @package gutenberg
  */
 
+if ( ! function_exists( 'get_default_block_editor_settings' ) ) {
+	/**
+	 * Gets the default block editor settings.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_Post $post Global post object.
+	 *
+	 * @return array The default block editor settings.
+	 */
+	function get_default_block_editor_settings( $post = null ) {
+		/**
+		 * Filters the allowed block types for the editor, defaulting to true (all
+		 * block types supported).
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param bool|array $allowed_block_types Array of block type slugs, or
+		 *                                        boolean to enable/disable all.
+		 * @param WP_Post    $post                The post resource data.
+		 */
+		$allowed_block_types = apply_filters( 'allowed_block_types', true, $post );
+
+		// Media settings.
+		$max_upload_size = wp_max_upload_size();
+		if ( ! $max_upload_size ) {
+			$max_upload_size = 0;
+		}
+
+		/** This filter is documented in wp-admin/includes/media.php */
+		$image_size_names = apply_filters(
+			'image_size_names_choose',
+			array(
+				'thumbnail' => __( 'Thumbnail' ),
+				'medium'    => __( 'Medium' ),
+				'large'     => __( 'Large' ),
+				'full'      => __( 'Full Size' ),
+			)
+		);
+
+		$available_image_sizes = array();
+		foreach ( $image_size_names as $image_size_slug => $image_size_name ) {
+			$available_image_sizes[] = array(
+				'slug' => $image_size_slug,
+				'name' => $image_size_name,
+			);
+		}
+
+		$default_size       = get_option( 'image_default_size', 'large' );
+		$image_default_size = in_array( $default_size, array_keys( $image_size_names ), true ) ? $image_default_size : 'large';
+
+		$image_dimensions = array();
+		$all_sizes        = wp_get_registered_image_subsizes();
+		foreach ( $available_image_sizes as $size ) {
+			$key = $size['slug'];
+			if ( isset( $all_sizes[ $key ] ) ) {
+				$image_dimensions[ $key ] = $all_sizes[ $key ];
+			}
+		}
+
+		$editor_settings = array(
+			'alignWide'              => get_theme_support( 'align-wide' ),
+			'allowedBlockTypes'      => $allowed_block_types,
+			'allowedMimeTypes'       => get_allowed_mime_types(),
+			'disableCustomColors'    => get_theme_support( 'disable-custom-colors' ),
+			'disableCustomFontSizes' => get_theme_support( 'disable-custom-font-sizes' ),
+			'disableCustomGradients' => get_theme_support( 'disable-custom-gradients' ),
+			'enableCustomLineHeight' => get_theme_support( 'custom-line-height' ),
+			'enableCustomSpacing'    => get_theme_support( 'custom-spacing' ),
+			'enableCustomUnits'      => get_theme_support( 'custom-units' ),
+			'isRTL'                  => is_rtl(),
+			'imageDefaultSize'       => $image_default_size,
```

Comment:
What happens when the function is actually defined in Core already and  we want to add new settings? (like imageDefaultSize here). Should we instead name the function more like `gutenberg_get_default_block_editor_settings` in the plugin and always manage to call no matter if it's defined on Core or not (acts as an override)?

Reply:
There is `block_editor_settings` filter that still can be used for that. The only remaining question is whether this function should absorb it as well.

Reply:
We can definitely use the filter for that (like now) but the reason I'm hesitant is that if we keep both functions it might make it easy to actually backport these settings to Core (just compare the functions). Not strongly opinionated though, whatever works best.

Now whether we include the filter in the function I'm not sure yet, specially because the filter is about all settings not just the default one.

Reply:
I copied all methods from the implementation I work on for WordPress core and prefixed all methods with `gutenberg_`. So it now follows the initial suggestion from @youknowriad.